src/Testcontainers.FakeGcsServer/FakeGcsServerBuilder.cs

```diff
@@ -0,0 +1,85 @@
+namespace Testcontainers.FakeGcsServer;
+
+/// <inheritdoc cref="ContainerBuilder{TBuilderEntity, TContainerEntity, TConfigurationEntity}" />
+[PublicAPI]
+public sealed class FakeGcsServerBuilder : ContainerBuilder<FakeGcsServerBuilder, FakeGcsServerContainer, FakeGcsServerConfiguration>
+{
+    public const string FakeGcsServerImage = "fsouza/fake-gcs-server:1.47.5";
+
+    public const ushort FakeGcsServerPort = 4443;
+
+    public const string StartupScriptFilePath = "/testcontainers.sh";
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="FakeGcsServerBuilder" /> class.
+    /// </summary>
+    public FakeGcsServerBuilder()
+        : this(new FakeGcsServerConfiguration())
+    {
+        DockerResourceConfiguration = Init().DockerResourceConfiguration;
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="FakeGcsServerBuilder" /> class.
+    /// </summary>
+    /// <param name="dockerResourceConfiguration">The Docker resource configuration.</param>
+    private FakeGcsServerBuilder(FakeGcsServerConfiguration dockerResourceConfiguration)
+        : base(dockerResourceConfiguration)
+    {
+        DockerResourceConfiguration = dockerResourceConfiguration;
+    }
+
+    /// <inheritdoc />
+    protected override FakeGcsServerConfiguration DockerResourceConfiguration { get; }
+
+    /// <inheritdoc />
+    public override FakeGcsServerContainer Build()
+    {
+        Validate();
+        return new FakeGcsServerContainer(DockerResourceConfiguration, TestcontainersSettings.Logger);
+    }
+
+    /// <inheritdoc />
+    protected override FakeGcsServerBuilder Init()
+    {
+        return base.Init()
+            .WithImage(FakeGcsServerImage)
+            .WithPortBinding(FakeGcsServerPort, true)
+            .WithEntrypoint("/bin/sh", "-c")
+            .WithCommand("while [ ! -f " + StartupScriptFilePath + " ]; do sleep 0.1; done; /bin/sh " + StartupScriptFilePath)
+            .WithWaitStrategy(Wait.ForUnixContainer().UntilMessageIsLogged("server started"))
+            .WithStartupCallback((container, ct) =>
+            {
+                const char lf = '\n';
+                var startupScript = new StringBuilder();
+                startupScript.Append("#!/bin/bash");
+                startupScript.Append(lf);
+                startupScript.Append("fake-gcs-server ");
+                startupScript.Append("-backend memory ");
+                startupScript.Append("-scheme http ");
+                // If we do not remove the trailing slash, uploading an object will result in an
+                // error: HttpStatusCode.NotFound. The HTTP request appears incorrect. The
+                // container logs indicate the presence of an extra slash: `PUT //upload/storage/v1`.
+                startupScript.Append("-external-url " + new UriBuilder(Uri.UriSchemeHttp, container.Hostname, container.GetMappedPublicPort(FakeGcsServerPort)).ToString().Trim('/'));
```

Comment:
It looks like you ran into this bug. I guess it is an issue in `fsouza/fake-gcs-server:1.47.5`. If the `-external-url` option ends with `/` you will receive 404 uploading an object.